### PR TITLE
Add documentation about merged cells virtualization feature

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR adds explanation to the documentation about a new merged cells virtualization feature. Also, the PR fixes the issue with an example checker that fails to run the puppeteer in the GH workflow.

_[skip changelog]_

### Related issue(s):
1. https://github.com/handsontable/handsontable/pull/11162

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

**Review by jansiegel — APPROVED** [review]

File: docs/content/guides/cell-features/merge-cells/merge-cells.md [hunk 1]
```diff
@@ -66,6 +66,58 @@ To initialize Handsontable with predefined merged cells, provide merged cells de
 
 :::
 
+## Optimizing rendering of the wide/tall merged cells
```
Comment (krzysztofspilka) [inline docs/content/guides/cell-features/merge-cells/merge-cells.md:69]: Maybe: Extremely wide and tall merged cells

File: docs/content/guides/cell-features/merge-cells/merge-cells.md [hunk 2]
```diff
@@ -66,6 +66,58 @@ To initialize Handsontable with predefined merged cells, provide merged cells de
...
+
+Sometimes, you might need to render a table with merged cells that are very wide or very tall. In such cases, the standard rendering mode may not be optimal. In these situations, we suggest enabling the virtual DOM for the merged cells.
```
Comment (krzysztofspilka) [inline docs/content/guides/cell-features/merge-cells/merge-cells.md:71]: Maybe: When cells span thousands of rows or columns, scrolling may feel slower compared to unmerged cells. To improve performance, consider enabling the dedicated virtualization feature for merged cells, which is disabled by default.

File: docs/content/guides/cell-features/merge-cells/merge-cells.md [hunk 3]
```diff
@@ -66,6 +66,58 @@ To initialize Handsontable with predefined merged cells, provide merged cells de
...
+```jsx
+mergeCells={{
+  virtualized: true,
+  cells: [{ row: 1, col: 1, rowspan: 200, colspan: 2 }]
+}}
+```
+
+:::
+
+The example below uses virtualized merged cells. It's also recommended to increase the buffer of rendered rows/columns to minimize the flickering effects.
```
Comment (krzysztofspilka) [inline docs/content/guides/cell-features/merge-cells/merge-cells.md:97]: Maybe: In the example below, virtualized merged cells are enabled. Additionally, increasing the buffer size for rendered rows and columns can help reduce flickering effects during scrolling.

File: docs/content/guides/cell-features/merge-cells/merge-cells.md (hunk 1 above)
Comment (budnix) [reply docs/content/guides/cell-features/merge-cells/merge-cells.md:69]: I would leave it as is. Those optimizations work not only for extremally wide/tall merged cells (e.g., row span > 1000) but also make sense for "regular" sizes (e.g., rowspan=50).

File: docs/content/guides/cell-features/merge-cells/merge-cells.md (hunk 2 above)
Comment (budnix) [reply docs/content/guides/cell-features/merge-cells/merge-cells.md:71]: The current text (after @jansiegel review) appeals to me more than what you suggested. I'd leave it as it is.

File: docs/content/guides/cell-features/merge-cells/merge-cells.md (hunk 3 above)
Comment (budnix) [reply docs/content/guides/cell-features/merge-cells/merge-cells.md:97]: Both, I'd say, sound the same .. I'd leave it as it is :)

**Review by jansiegel — APPROVED** [review]

**Pull request merged**